plugins/kubernetes/app/models/kubernetes/template_filler.rb

```diff
@@ -195,12 +194,13 @@ def prefix_service_cluster_ip
 
     def set_namespace
       return if template[:metadata].key?(:namespace)
-      template[:metadata][:namespace] = @doc.deploy_group.kubernetes_namespace
+      template[:metadata][:namespace] = project.kubernetes_namespace&.name || @doc.deploy_group.kubernetes_namespace
```

Comment:
i think `||=` would be better...

Reply:
can do as a separate PR, but we _should_ do it

Reply:
I can't thin of a scenario where that would be better, atm we validate that users are using their namespace or namespace-less resources ... if we do `||=` then that adds more ambiguity as to what "nil" namespace means and we again override a configured setting